examples/http_proxy.rs

```diff
@@ -1,7 +1,7 @@
 #![deny(warnings)]
 
 use std::convert::Infallible;
-use std::net::SocketAddr;
+use std::net::{SocketAddr, ToSocketAddrs};
```

Comment:
Tokio now has an asynchronous DNS resolution function: https://docs.rs/tokio/0.2.8/tokio/net/fn.lookup_host.html. Maybe consider using that instead, as std's `to_socket_addrs()` is a blocking operation.

Reply:
Or another option, that doesn't require enabling another Tokio feature, is to just use `hyper::client::connect::dns::GaiResolver`.

Reply:
> Tokio now has an asynchronous DNS resolution function: https://docs.rs/tokio/0.2.8/tokio/net/fn.lookup_host.html. Maybe consider using that instead, as std's `to_socket_addrs()` is a blocking operation.

it seems we need upgrade `tokio` to use `lookup_host`.

I  `pub` the  `resolve` method in `dns.rs`, is it appropriate?